packages/cactus-plugin-persistence-fabric/src/main/typescript/plugin-persistence-fabric.ts

```diff
@@ -0,0 +1,914 @@
+        chaincodeID =
+          transactionDataObject.payload.data.actions[0].payload.action
+            .proposal_response_payload.extension;
+        status =
+          transactionDataObject.payload.data.actions[0].payload.action
+            .proposal_response_payload.extension.response.status;
+        this.log.info("rwset  :", JSON.stringify(rwset));
+        rwset =
+          transactionDataObject.payload.data.actions[0].payload.action
+            .proposal_response_payload.extension.results.ns_rwset;
+
+        if (rwset !== undefined) {
+          readSet = rwset.reads;
+          writeSet = rwset.writes;
+          read_set = JSON.stringify(readSet, null, 2);
+          write_set = JSON.stringify(writeSet, null, 2);
+        } else {
+          readSet = " as usual ";
+          writeSet = " as usual ";
+        }
+
+        chaincode_proposal_input =
+          transactionDataObject.payload.data.actions[0].payload
+            .chaincode_proposal_payload.input.chaincode_spec.input.args;
+        if (chaincode_proposal_input !== undefined) {
+          let inputs = "";
+          for (const input of chaincode_proposal_input) {
+            inputs =
+              (inputs === "" ? inputs : `${inputs},`) +
+              Buffer.from(input).toString("utf8");
+          }
+          chaincode_proposal_input = inputs;
+        }
+        endorser_signature =
+          transactionDataObject.payload.data.actions[0].payload.action
+            .endorsements[0].signature;
+        if (endorser_signature !== undefined) {
+          endorser_signature = Buffer.from(endorser_signature).toString("hex");
+        }
+        payload_proposal_hash = transactionDataObject.payload.data.actions[0].payload.action.proposal_response_payload.proposal_hash.data.toString(
+          "hex",
+        );
+        endorser_id_bytes = transactionDataObject.payload.data.actions[0].payload.action.endorsements[0].endorser.id_bytes.data.toString(
+          "hex",
+        );
+
+        end_mspid =
+          tempBlockParse.decodedBlock.data.data[txIndex].payload.data.actions[0]
+            .payload.action.endorsements[0].endorser.mspid;
+        tx_response =
+          tempBlockParse.decodedBlock.data.data[txIndex].payload.data.actions[0]
+            .payload.action.proposal_response_payload.extension;
+      }
+
+      if (
+        transactionDataObject.payload.header.channel_header.typeString ===
+        "CONFIG"
+      ) {
+        txid = sha256(transactionDataStringifies);
+        readSet =
+          transactionDataObject.payload.data.last_update.payload?.data
+            .config_update.read_set;
+        writeSet =
+          transactionDataObject.payload.data.last_update.payload?.data
+            .config_update.write_set;
+      } else {
+        txid =
+          tempBlockParse.decodedBlock.data.data[txIndex].payload.header
+            .channel_header.tx_id;
+      }
+
+
+      const chaincode_id: string = JSON.stringify(chaincodeID);
```

Comment:
This should only be executed if we have a valid chaincodeID, because according to this code, chaincodeID only exists when `transactionDataObject.payload.data.actions !== undefined`

And then if `chaincode_id` is valid only for this above condition, then function calls like `insertDetailedTransactionEntry` should also be done only if the above condition is satisfied / exception is handled.

Reply:
insertDetailedTransactionEntry  is always executed , there are transactions like config transactions and not every transaction is related to chaincodeID, but will recheck that

Reply:
insertDetailedTransactionEntry is always executed , there are transactions like config transactions and not every transaction is related to chaincodeID, but will recheck that